src/utils/session.ts

```diff
@@ -22,3 +22,26 @@ export function getPresenterAccountDetails(req: Request): Details {
 export function setPresenterAccountDetails(req: Request, details: Details) {
     req.session?.setExtraData(PRESENTER_ACCOUNT_SESSION_KEY, details);
 }
+
+export function generateUserDetails(req: Request): Details{
+    const signinInfo = req.session?.data?.signin_info;
```

Comment:
Since all of the subsequent calls reference the `user_profile` object it would be better to pull that object out into a variable here intead of just getting the `signin_info`.

Additionally, the call chain may result in an undefined value. This needs to be checked and handled.